doc/control-plane-events.md

```diff
@@ -0,0 +1,175 @@
+# Control Plane Events
+
+## Introduction
+Control plane events have been added to enable the device service to emit events onto the message bus when a device has been added, updated, or deleted. This is in lieu of official Control Plane Events (CPE) in an upcoming release of EdgeX. Once EdgeX officially supports CPE, these events will be dropped in favor of those.
+
+See edgex-docs [issue #581](https://github.com/edgexfoundry/edgex-docs/issues/581)
+
+## Sample Events
+The following are sample events emitted when a camera is added, updated or deleted from the device service.
+
+### CameraAdded
+The following is a sample event emitted when a camera is added to the device service. The top level object is an EdgeX `Event`, which contains 1 or more `Readings`.
+Where each `Reading` contains the following information:
+- **`origin`**: The timestamp at which the event occurred (in nanoseconds since Epoch)
+- **`resourceName`**: The type of control plane event (`CameraAdded`)
+- **`objectValue`**: The entire EdgeX Device object
+  - **`Name`**: The newly added device's name
+  - **`ProfileName`**: Which EdgeX Profile the device was assigned
+  - **`Protocols.Onvif`**: Onvif specific connection information
+
+```json
+{
+   "apiVersion":"v2",
+   "id":"2b20ddfb-d6b2-4e63-b7dd-0abef38524dd",
+   "deviceName":"device-onvif-camera",
+   "profileName":"device-onvif-camera",
+   "sourceName":"CameraAdded",
+   "origin":1650895116540236500,
+   "readings":[
+      {
+         "id":"5bf9958b-c504-4803-824d-5796971a3128",
+         "origin":1650895116540236500,
+         "deviceName":"device-onvif-camera",
+         "resourceName":"CameraAdded",
+         "profileName":"device-onvif-camera",
+         "valueType":"Object",
+         "objectValue":{
+            "AdminState":"UNLOCKED",
+            "AutoEvents":[
+               
+            ],
+            "Created":0,
+            "Description":"Intel SimCamera Camera",
+            "Id":"",
+            "Labels":[
+               "auto-discovery",
+               "Intel",
+               "SimCamera"
+            ],
+            "LastConnected":0,
+            "LastReported":0,
+            "Location":null,
+            "Modified":0,
+            "Name":"Intel-SimCamera-c45a57b3-6fcb-4c51-83de-54495e5454a6",
+            "Notify":false,
+            "OperatingState":"UP",
+            "ProfileName":"onvif-camera",
+            "Protocols":{
+               "Onvif":{
+                  "Address":"172.20.25.54",
+                  "AuthMode":"usernametoken",
+                  "FirmwareVersion":"2.4a",
+                  "HardwareId":"1.0",
+                  "Manufacturer":"Intel",
+                  "Model":"SimCamera",
+                  "Port":"10000",
+                  "SecretPath":"credentials002",
+                  "SerialNumber":"c45a57b3"
+               }
+            },
+            "ServiceName":"device-onvif-camera"
```

Comment:
Is all this data needed? or can we just have the new device name as a simple string value? 
```
         "valueType":"String",
         "value":"Intel-SimCamera-c45a57b3-6fcb-4c51-83de-54495e5454a6"
```
If not, what are we using all this info for. If the receiver need more data they can query Core Metadata for the details.
I assume for Skyline trail we just device name to use in Core Command that is needed.

Reply:
Originally that is what we had, but I decided to send the whole device in case they needed it. Thinking back over it, you are right, they can just query the metadata service for it. My initial hesitation from that was probably linked to an issue we had in the past with the LLRP service, but this one will not suffer from that due to the events being sent _after_ the device service cache is updated.